backend/src/zango/test/client.py

```diff
@@ -0,0 +1,90 @@
+from importlib import import_module
+
+from django.conf import settings
+from django.contrib.auth import authenticate
+from django.http import HttpRequest, SimpleCookie
+from django.test import Client, RequestFactory
+
+from zango.middleware.tenant import ZangoTenantMainMiddleware
+
+
+class BaseZangoRequestFactory:
+    tm = ZangoTenantMainMiddleware(lambda r: r)
+
+    def __init__(self, tenant, **defaults):
+        super().__init__(**defaults)
+        self.tenant = tenant
+        self.user = None
+
+    def generic(self, *args, **kwargs):
+        if "HTTP_HOST" not in kwargs:
+            kwargs["HTTP_HOST"] = self.tenant.get_primary_domain().domain
+        request = super().generic(*args, **kwargs)
+        # Assign the tenant to the request object
+        request.tenant = self.tenant
+        if request.tenant.tenant_type == "app":
+            request.user = self.user
+        return request
+
+
+class ZangoRequestFactory(BaseZangoRequestFactory, RequestFactory):
+    pass
+
+
+class ZangoClient(BaseZangoRequestFactory, Client):
+    def _create_mock_request(self):
```

Comment:
@DevilsAutumn in `zango.core.utils` we have `get_mock_request`. We can check if we can utilize it for creating mock requests in tests

Reply:
we can update `get_mock_request` as well, it has kwargs so session can be passed in kwargs.